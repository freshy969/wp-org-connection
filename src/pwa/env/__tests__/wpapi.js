import wpapi from '../wpapi';

describe('Connection â€º WpApi', () => {
  test('should init without throwing', async () => {
    expect(() => wpapi.init({ siteUrl: 'https://example.com' })).not.toThrow();
  });
  test('should throw if no url is present', async () => {
    expect(() => wpapi.init()).toThrow();
  });
  test('should build correct urls for getEntity', async () => {
    wpapi.init({ siteUrl: 'https://example.com' });
    expect(wpapi.getEntity({ type: 'post', id: 60 }).toString()).toMatchSnapshot();
    expect(wpapi.getEntity({ type: 'media', id: 62 }).toString()).toMatchSnapshot();
    expect(wpapi.getEntity({ type: 'category', id: 7 }).toString()).toMatchSnapshot();
    expect(wpapi.getEntity({ type: 'tag', id: 10 }).toString()).toMatchSnapshot();
    expect(wpapi.getEntity({ type: 'page', id: 3 }).toString()).toMatchSnapshot();
  });
  test('should build correct urls for getEntity using custom post types', async () => {
    wpapi.init({
      siteUrl: 'https://example.com',
      cptEndpoints: {
        session: 'sessions',
        speaker: 'speakers',
      },
    });
    expect(wpapi.getEntity({ type: 'session', id: 60 }).toString()).toMatchSnapshot();
    expect(wpapi.getEntity({ type: 'speaker', id: 62 }).toString()).toMatchSnapshot();
  });
  test('should throw if custom post type is not registered', async () => {
    wpapi.init({ siteUrl: 'https://example.com' });
    expect(() => wpapi.getEntity({ type: 'session', id: 60 }).toString()).toThrow();
  });
  test('should build correct urls for getListPage', async () => {
    wpapi.init({ siteUrl: 'https://example.com' });
    expect(
      wpapi.getListPage({ type: 'latest', id: 'post' }).toString(),
    ).toMatchSnapshot();
    expect(
      wpapi.getListPage({ type: 'latest', id: 'post', page: 2 }).toString(),
    ).toMatchSnapshot();
    expect(
      wpapi.getListPage({ type: 'category', id: 7, page: 3, perPage: 5 }).toString(),
    ).toMatchSnapshot();
    expect(
      wpapi.getListPage({ type: 'tag', id: 10, page: 10, perPage: 3 }).toString(),
    ).toMatchSnapshot();
  });
  test('should build correct urls for getCustom', async () => {
    wpapi.init({ siteUrl: 'https://example.com' });
    expect(
      wpapi.getCustom({ type: 'post' }).toString(),
    ).toMatchSnapshot();
    expect(
      wpapi.getCustom({ type: 'category', params: { includes: 7 }, page: 2 }).toString(),
    ).toMatchSnapshot();
    expect(
      wpapi.getCustom({ type: 'page', params: { includes: '3,4' }, page: 1 }).toString(),
    ).toMatchSnapshot();
  });
});
