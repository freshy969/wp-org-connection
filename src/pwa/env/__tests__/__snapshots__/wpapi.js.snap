// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection › WpApi should build correct urls for getCustom 1`] = `"https://example.com/?rest_route=/wp/v2/posts&_embed=true&page=1"`;

exports[`Connection › WpApi should build correct urls for getCustom 2`] = `"https://example.com/?rest_route=/wp/v2/categories&_embed=true&includes=7&page=2"`;

exports[`Connection › WpApi should build correct urls for getCustom 3`] = `"https://example.com/?rest_route=/wp/v2/pages&_embed=true&includes=3%2C4&page=1"`;

exports[`Connection › WpApi should build correct urls for getEntity 1`] = `"https://example.com/?rest_route=/wp/v2/posts/60&_embed=true"`;

exports[`Connection › WpApi should build correct urls for getEntity 2`] = `"https://example.com/?rest_route=/wp/v2/media/62&_embed=true"`;

exports[`Connection › WpApi should build correct urls for getEntity 3`] = `"https://example.com/?rest_route=/wp/v2/categories/7&_embed=true"`;

exports[`Connection › WpApi should build correct urls for getEntity 4`] = `"https://example.com/?rest_route=/wp/v2/tags/10&_embed=true"`;

exports[`Connection › WpApi should build correct urls for getEntity 5`] = `"https://example.com/?rest_route=/wp/v2/pages/3&_embed=true"`;

exports[`Connection › WpApi should build correct urls for getEntity using custom post types 1`] = `"https://example.com/?rest_route=/wp/v2/sessions/60&_embed=true"`;

exports[`Connection › WpApi should build correct urls for getEntity using custom post types 2`] = `"https://example.com/?rest_route=/wp/v2/speakers/62&_embed=true"`;

exports[`Connection › WpApi should build correct urls for getListPage 1`] = `"https://example.com/?rest_route=/wp/v2/posts&_embed=true&page=1&per_page=10"`;

exports[`Connection › WpApi should build correct urls for getListPage 2`] = `"https://example.com/?rest_route=/wp/v2/posts&_embed=true&page=2&per_page=10"`;

exports[`Connection › WpApi should build correct urls for getListPage 3`] = `"https://example.com/?rest_route=/wp/v2/posts&_embed=true&categories=7&page=3&per_page=5"`;

exports[`Connection › WpApi should build correct urls for getListPage 4`] = `"https://example.com/?rest_route=/wp/v2/posts&_embed=true&page=10&per_page=3&tags=10"`;
